### ISSUE TITLE
ability to set annotations+labels on pods created by argocd

### ISSUE THREAD

**Issue by Morriz** [argoproj-labs/argocd-operator#637]
**Is your feature request related to a problem? Please describe.**
Hi, we need to set annotations on any pods running on our cluster, but I can't find the option to do so on pods deployed by argo.

**Describe the solution you'd like**
Something like:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
...
controller:
  podAnnotations:
    some: anno
  labels:
    some: label
...
server:
  podAnnotations:
    ana: da
```

**Comment by iam-veeramalla** [comment]
@Morriz I agree it is very useful, but I see a concern with operator reconciling the labels and annotations. There is a good chance that operator may removes annotations/labels set by admission controllers. WDYT ?

**Comment by Morriz** [comment]
> @Morriz I agree it is very useful, but I see a concern with operator reconciling the labels and annotations. There is a good chance that operator may removes annotations/labels set by admission controllers. WDYT ?

That is a matter of having a maature operator that allows for labels it does not manage. That is why one namespaces them, in order to filter.

**Comment by iam-veeramalla** [comment]
LGTM. Would you like to work on this one ? If not I can add the label `help-wanted` and remove `traige-required`.

Thanks @Morriz

**Comment by Morriz** [comment]
I am no Go dev unfortunately, and my calendar is rather full ;|

**Comment by svghadi** [comment]
https://github.com/argoproj-labs/argocd-operator/pull/1532 implemented support to add custom annotations and labels on Argo CD component pods created by Operator. New fields are introduced in ArgoCD CR to support this. Refer [custom_labels_annotations.md ](https://github.com/argoproj-labs/argocd-operator/blob/master/docs/usage/custom_labels_annotations.md) for more details.

It will be available in upcoming 0.13.0 release .
